Tests/UnitTests/Mocks/MockStoreKit2TransactionListener.swift

```diff
@@ -62,10 +62,17 @@ final class MockStoreKit2TransactionListener: StoreKit2TransactionListenerType {
         self.invokedHandleParameters = (.init(purchaseResult), ())
         self.invokedHandleParametersList.append((.init(purchaseResult), ()))
 
-        let transaction: StoreTransaction? = self.mockTransaction.value.map {
-            StoreTransaction(sk2Transaction: $0,
-                             jwsRepresentation: self.mockJWSToken,
-                             environmentOverride: self.mockEnvironment)
+        var transaction: StoreTransaction
+
+        if let mockTransaction = self.mockTransaction.value {
+            transaction = StoreTransaction(sk2Transaction: mockTransaction,
+                                           jwsRepresentation: self.mockJWSToken,
+                                           environmentOverride: self.mockEnvironment)
+        } else {
+            let result = purchaseResult.verificationResult!
+            transaction = StoreTransaction(sk2Transaction: result.underlyingTransaction,
+                                           jwsRepresentation: result.jwsRepresentation,
+                                           environmentOverride: self.mockEnvironment)
```

Comment:
In order to properly test this, we need to return the actual transaction made by the orchestrator, instead of providing a mock.

It cannot be tested with `invokedHandleParameters` because, if we have set a mock transaction using `simulateAnyTransaction`, the orchestrator would return the same mocked transaction:

![image](https://github.com/RevenueCat/purchases-ios/assets/1014118/c262a0ed-5dab-4413-bd1b-14da9ec200a6)